### ISSUE TITLE
Iphone App Login Issue

### ISSUE THREAD

**Issue by bap72190** [twilio/OpenVBX-iPhone#11]
I am running Openvbx on a vps server with Cpanel. Can't seem to get the Iphone app to work, just says login is invalid, even though I am giving it the correct login. Everything else works fine through a browser. I have tried all of the suggestions that are posted online and still seem to have no success. I do have root access to my server if that helps at all. Also, app seems to randomly crash when you click next after entering the URL.

**Comment by yshuman** [comment]
im having the same problem

**Comment by jefflinwood** [comment]
I just ran through this exact problem, and it was because I was using a shared host, and the rewrite rules in .htaccess don't work well - it's not a problem with the iPhone app, but a problem with the PHP server. I was able to confirm this by repeating the exact calls through the Firefox REST Client.

I installed OpenVBX on a VPS Ubuntu server, and the iOS app worked.

**Comment by jrstarke** [comment]
@jefflinwood Did you ever figure out what in the rewrite rules doesn't work on a shared host?

**Comment by jefflinwood** [comment]
@jrstarke No, I spent a decent amount of time on it, but it was easier for me to set it up on a VPS than to continue trying to debug the authentication with .htaccess - I did end up trying most of the possible solutions here.

**Comment by jrstarke** [comment]
Thanks @jefflinwood. I managed to figure it out on my end. 

For anyone else facing this problem, I took a suggestion from https://github.com/twilio/OpenVBX/issues/122#issuecomment-4014787 to log the incoming requests. I found in my case that although the system supports changing `REDIRECT_AUTHORIZATION` back, for some reason, my AUTHORIZATION was in REDIRECT_REDIRECT_AUTHORIZATION. After adding a small modification to the source code, it worked.

**Comment by JeffaCubed** [comment]
Just to add a bit of back-story to the login + crashing issues with the OpenVBX iOS app. I'd wondered for quite some time why the iOS app stopped working ~ iOS 3-4. I'd tried hosting OpenVBX on shared hosting w/ HostGator + Site5, a managed VPS on Site5,  with all functional working except the systemic iOS app crashing, login issues, etc. I decided to take another crack at the iOS issue recently (late 2013), this time setting up OpenVBX (1.2.13) as an Amazon AWS EC2 instance (ubuntu 12.0.4 - Free Tier). A few immediate observations, which may mirror a bit of what @jefflinwood found when testing OpenVBX on a VPS Ubuntu server:
1. OpenVBX is extremely fast + stable even on the free EC2 (ubunto 12.0.4) tier (compared to shared &/or managed VPS hosting). I'd say it makes zero sense to host OpenVBX on shared hosting
2. For the first time in years, the iOS app for OpenVBX (even though it is very outdated at this point & still relies on PSTN vs Twilio Client) does work 100% (as far as I can tell) on AWS EC2. For example, incoming + outgoing text messages refresh instantly on the iOS app (no crashing, immediate + valid refreshing for new sms messages)
3. Enabling mod_rewrite in the settings will cause OpenVBX to stop working while hosted on EC2 (until one goes back in the db & sets mod_rewrite to 'off'). I'm sure this is related to having not enabled mod_rewrite correctly on the EC2 instance - though I'm open to suggestions as to why mod_rewrite does not work on EC2!
4. Outgoing voicemail/sms alert emails + password recovery emails clearly do not work w/ OpenVBX hosted on EC2 - but again that likely related to not configuring outgoing email on the EC2 instance (which I'm not exactly sure how to do - yet) Open to suggestions on the EC2 email piece for OpenVBX

Here's the resource I used to install OpenVBX on EC2 > http://sidielazhari.blogspot.ca/2013/10/how-to-install-openvbx-on-amazon-ec2.html?m=1 (by sidielazhari)

Cheers,

-Jeff (Vancouver)

**Comment by olea555** [comment]
Quick fix for this problem if using EC2 and may apply to other servers. The problem does seem to be with mod_rewrite. It appears both Iphone and Android App requires the mod_rewrite to be working. 

On EC2 the modrewrite was loaded by default, but you have to check the following:

Check your etc/httpd/conf/httpd.conf file. It should have the following in it:

AllowOverride All

Basically, you’ll be adding All insted of None to AllowOverride, you do not want to edit the main directory configuration, you want to edit the one that looks like this:

<Directory “/var/www/html”>

Not:

<Directory />

Then restart Apache with:

sudo service httpd restart

from the command line.

Credits to jafty.com who pointed me in the right direction with: http://jafty.com/blog/enable-mod_rewrite-on-apache-ec2-linux-server/
